packages/organizations/origin-organization-irec-api/src/beneficiary/handlers/create-beneficiary.handler.ts

```diff
@@ -0,0 +1,40 @@
+import { Repository } from 'typeorm';
+import { CommandBus, CommandHandler, ICommandHandler } from '@nestjs/cqrs';
+import { InjectRepository } from '@nestjs/typeorm';
+import { UserService } from '@energyweb/origin-backend';
+
+import { IrecService } from '../../irec';
+import { CreateBeneficiaryCommand } from '../commands/create-beneficiary.command';
+import { Beneficiary } from '../beneficiary.entity';
+import { BeneficiaryDTO } from '../dto/beneficiary.dto';
+import { GetBeneficiaryCommand } from '../commands';
+
+@CommandHandler(CreateBeneficiaryCommand)
+export class CreateBeneficiaryHandler implements ICommandHandler<CreateBeneficiaryCommand> {
+    constructor(
+        @InjectRepository(Beneficiary)
+        private readonly repository: Repository<Beneficiary>,
+        private readonly userService: UserService,
+        private readonly irecService: IrecService,
+        private readonly commandBus: CommandBus
+    ) {}
+
+    public async execute({ organization }: CreateBeneficiaryCommand): Promise<BeneficiaryDTO> {
+        const platformAdmin = await this.userService.getPlatformAdmin();
+
+        const irecBeneficiary = await this.irecService.createBeneficiary(
+            platformAdmin.id,
+            organization
+        );
+
+        const beneficiary = this.repository.create({
+            irecBeneficiaryId: irecBeneficiary.id,
+            organizationId: organization.id,
+            ownerOrganizationId: platformAdmin.organization.id
```

Comment:
Hey @ioncreature what's the idea behind that ownerOrganizationId field? Seems like that's always set to a platform owner organization.

Reply:
We create a beneficiary for each new organization for the platform account.
But different organization will create their own subset of beneficiaries based on this field

Reply:
`platformAdmin.organization.id` is always the same organization, like PTT or FOTON. Did you maybe meant to use `loggedInUser.organizationId` instead?

Reply:
I think `loggedInUser.organizationId` is stored in the line above as `organizationId`.

@ioncreature if `platformAdmin.organization.id` is needed, I assume there no need to store it in the database with every row, as it will always stay the same throughout the life-cycle of the application?

Can't it be fetched only when we interact with the I-REC wrapper?

Reply:
@ioncreature I think we do a little bit of guessing here, could you please explain what each of these fields relates to?

Reply:
@JosephBagaric Yep, agree. We can get rid of it. I will set `null` instead of `platformAdmin. organization.id`. And it will mean the same

Reply:
You can remove the entire column, no sense in keeping a column that's always `null`.

Reply:
`irecBeneficiaryId` and `organizationId` identify IREC beneficiary and connected origin organization. So they are always together.

`ownerId` (was `ownerOrganizationId` before) is to show who is the owner.

It means that everybody can have read access to the platform beneficiaries list, but if someone wants to use beneficiary then he has to add this beneficiary to his own organization list. And then use.

Reply:
https://github.com/energywebfoundation/origin/pull/2482#discussion_r648078093
Yes, we create all IREC beneficiaries from the platform account

Reply:
So we have 2 different handlers - `create beneficiary` which creates beneficiary for org using platform account. 
And `add beneficiary` which adds it to the organization beneficiaries list

Reply:
Given all that we discussed, I think the database table is correct so you store `irecBeneficiaryId`, `organizationId`, and `ownerId`. Since looks like IREC API beneficiary is bound to the IREC organization, you need to follow that principle as well here, so 

- each platform organization (ownerId) maintains it's own list of beneficiaries and the list is private to them
- during redemption (claiming) the organization who requests uses the `irecBeneficiaryId` that is created by them as `beneficiary_id` in https://evident-registry.atlassian.net/wiki/spaces/EV/pages/216531743/POST%2Bapi%2Birec%2Bv1%2Bredemption-management

Reply:
@kosecki123 It is already it. Except the redemption - it is on different task